src/activeobjectmgr.h

```diff
@@ -52,24 +51,16 @@ class ActiveObjectMgr
 
 	void clear()
 	{
-		while (!m_active_objects.empty())
-			removeObject(m_active_objects.begin()->first);
+		for (auto &it : m_active_objects.iter()) {
+			if (!it.second)
+				continue;
+			m_active_objects.remove(it.first);
+		}
```

Comment:
`on_destruct` can add objects. `clear()` should make `m_active_objects` empty (or crash if not possible?). This is why the loop was made with `.empty()`. (I should've put a comment there.)

Reply:
Ah so that is the entire reason. I indeed didn't know.